funcbench/README.md

```diff
@@ -28,8 +28,53 @@ The benchmark can be triggered by creating a comment in a PR which specifies a b
 /funcbench <branch/commit> <benchmark function regex>
 ```
 
+#### Setup
+
 The comment is handled by [comment-monitor](https://github.com/prometheus/test-infra/tree/master/tools/commentMonitor) and then the parsed arguments are handed over to funcbench(if using Github Actions) or to [prombench](https://github.com/prometheus/test-infra/tree/master/prombench) if using funcbench with GKE.
 
+- Create GitHub actions workflow file (see below) that is executed when an `repository_dispatch` event is on.
+- Read BRANCH / PR_NUMBER / REGEX from event payload into environment variables.
+
+#### Example GitHub action workflow file
+
+```yaml
+on: repository_dispatch
+name: Funcbench Workflow
+jobs:
+  run_funcbench:
+    name: Running funcbench
+    if: github.event.action == 'funcbench_start'
+    runs-on: ubuntu-latest
+    env:
+      AUTH_FILE: ${{ secrets.PROMBENCH_GKE_AUTH }}
+      CLUSTER_NAME: << cluster name >>
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      PROJECT_ID: << project id >>
+      PR_NUMBER: ${{ github.event.client_payload.PR_NUMBER }}
+      TEST_INFRA_REPO: https://github.com/prometheus/test-infra.git # put your repo here if test your own test-infra
+      ZONE: << zone >>
+    steps:
+    - name: Prepare nodepool
+      uses: docker://prominfra/prombench:latest
+      with:
+        args: make funcbench_nodepool_create
+    - name: Run funcbench
+      uses: docker://prominfra/prombench:latest
+      env:
+        BRANCH: ${{ github.event.client_payload.BRANCH }}
+        GITHUB_ORG: prominfra
+        GITHUB_REPO: prometheus
+        GITHUB_TOKEN: ${{ secrets.PERSONAL_TOKEN }} # if benchmark lasts more than 60 minutes, have to create your own token, and put it into PERSONAL_TOKEN
+        REGEX: ${{ github.event.client_payload.REGEX }}
+      with:
+        args: make funcbench_resource_apply
+    - name: Recycle all
+      if: always()
```

Comment:
is that nececary?

Reply:
see the next comments.

Reply:
sorry I am not sure which comments are you referring to here?

Reply:
I'm referring to [this](https://github.com/prometheus/test-infra/pull/343#discussion_r408710386). Line 186 `_ = c.jobDelete(resource)` is already removed from code.
So we have to remove the `Job` resource everytime it finishes, no matter it's successful or failed.

Reply:
no I was referring only to the `if: always()` line. Would this execute always even without this line?

Reply:
Yes, we can remove this line. It just emphasizes that it's always executed. In real case, we will have `if: failure()` and `if: success()` behind. Using `if: always()` makes sure that the exit code of step `Run funcbench` can be passed down the chain (not interrupted by the exit code of step `Recycle all`).